### PR TITLE
DSND-1879 Add disqualified search routes to UserAuthorisationInterceptor

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Change UserAuthorisationInterceptor to accept DELETE requests
* Tested locally and can confirm functionality works as expected.

Resolves [DSND-1879](https://companieshouse.atlassian.net/browse/DSND-1879)

[DSND-1879]: https://companieshouse.atlassian.net/browse/DSND-1879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sthompsonCH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Review by cgregoryCH — COMMENTED** [review]
Looks good, sonar reporting a missed test condition around the interceptor logic. I think maybe a test that a GET is not allowed would add to the coverage.

File: src/main/java/uk/gov/companieshouse/search/api/interceptor/UserAuthorisationInterceptor.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ private boolean validateAPI(HttpServletRequest request, HttpServletResponse resp
         DataMap.Builder builder = new DataMap.Builder()
                 .requestId(request.getHeader(REQUEST_ID_HEADER_NAME));
 
-        if(AuthorisationUtil.hasInternalUserRole(request) && PUT.matches(request.getMethod())) {
+        if(AuthorisationUtil.hasInternalUserRole(request) && !GET.matches(request.getMethod())) {
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/search/api/interceptor/UserAuthorisationInterceptor.java:39]: Will this block a health check request that will normally be a `GET` request?

Comment (cgregoryCH) [reply src/main/java/uk/gov/companieshouse/search/api/interceptor/UserAuthorisationInterceptor.java:39]: I think adding some unit tests for GET requests without different types of authentication would help

Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/search/api/interceptor/UserAuthorisationInterceptor.java:39]: GET requests unit tests added in latest commit
@johncook-ch as the healthcheck route is not added to the registry, it wouldn't be affected.

**Pull request merged**